lib/castle_devise/patches/passwords_controller.rb

```diff
@@ -8,6 +8,52 @@ module Patches
     module PasswordsController
       extend ActiveSupport::Concern
 
+      # PUT /resource/password
+      # @note Notice, all these steps must happen within a block because before calling
+      #   "reset_password_by_token" method we don't know what resource we operate on.
+      def update
+        super do |resource|
+          next unless resource_class.castle_hooks[:profile_update]
+
+          context = CastleDevise::Context.from_rack_env(request.env, scope_name, resource)
+
+          begin
+            # No need for a verification mechanism here because this action is performed after
+            # clicking on the password reset link with a token.
+            CastleDevise.sdk_facade.risk(
+              event: "$profile_update",
+              status: "$attempted",
+              context: context
+            )
```

Comment:
Actually, there is no sense in making this call as we cannot perform this outside of the block, before entering the super method because we don't know what resource we operate on then and performing this in the block doesn't make sense because it would be performed after [this action happened](https://github.com/heartcombo/devise/blob/c82e4cf47b02002b2fd7ca31d441cf1043fc634c/lib/devise/models/recoverable.rb#L145-L161) so already after resetting the password.